leftwm-core/src/display_servers/xlib_display_server/mod.rs

```diff
@@ -175,22 +174,50 @@ impl DisplayServer for XlibDisplayServer {
                 self.xw.set_state(h, toggle_to, state);
                 None
             }
-            DisplayAction::SetWindowOrder(wins) => {
-                // get all the windows are aren't managing.
-                // They should be in front of our windows.
-                let unmanged: Vec<WindowHandle> = self
+            DisplayAction::SetWindowOrder(windows) => {
+                // The windows we are managing should be behind unmagaed windows. Unless they are
+                // fullscreen, or their children.
+                let (fullscreen_windows, other): (Vec<&Window>, Vec<&Window>) =
+                    windows.iter().partition(|w| w.is_fullscreen());
+                // Fullscreen windows.
+                let level2: Vec<WindowHandle> =
+                    fullscreen_windows.iter().map(|w| w.handle).collect();
+                let (fullscreen_children, other): (Vec<&Window>, Vec<&Window>) =
+                    other.iter().partition(|w| {
+                        level2.contains(&w.transient.unwrap_or(WindowHandle::XlibHandle(0)))
+                    });
+                // Fullscreen windows children.
+                let level1: Vec<WindowHandle> =
+                    fullscreen_children.iter().map(|w| w.handle).collect();
+
+                log::info!("Children: {:?}", level1);
```

Comment:
Do we need such `info` logs in master? Maybe it should be `debug!` ?

Reply:
Forgot to remove them.